### ISSUE TITLE
Can I skip 'Intro CS' courses?

### ISSUE THREAD

**Issue by cauequeiroz** [ossu/computer-science#745]
hey guys,

Is it okay to skip all 'Intro CS' courses and jump directly to 'Core CS'?

I already know that I want to learn computer science and, as a professional developer since 2013, I have some programming experience. Are these courses only for beginners who want to try a new area to see if they like it or do these courses have some knowledge that I will not see in Core CS courses?

**Comment by spamegg1** [comment]
Intro CS has two SUBsections: Introduction to Programming and Introduction to Computer Science.

What you described fits the "Introduction to Programming" subsection of Intro CS, with Python for Everyone and Fundamentals of Computing: 

> If you've never written a for-loop, or don't know what a string is in programming, start here. Choose one of the two course series below. Either one will give you an introduction to programming that assumes no prior knowledge.

If you have programming experience you can skip these if you want.

But if you've never had any CS before, don't skip the "Introduction to Computer Science" subsection, with Introduction to Programming and Computer Science with Python (I'll abbreviate this as MITx6.00.1x). I'm sure you'll find it relatively easy. 

Some of the topics in that course are covered in Core CS but in a more advanced way; skipping this course and jumping straight into Core CS can make it difficult to understand them when they are covered there.

An alternate and much more challenging approach would be to skip Intro CS completely and jump into Core CS. In the past Harvard's CS50 (now in Core Systems) was the very first  course on the curriculum, back when I started. That course is a much harder "introduction" than MITx6.00.1x, but back then we all did it that way. 

However there is a reason that these Intro CS courses were added to the curriculum. Personally I'm not a big fan of the Intro to Programming subsection of Intro CS (I find it redundant, it's covered in MITx6.00.1x), but MITx6.00.1x is a much gentler introduction (while still having a decent challenge), which probably makes much more sense for most beginners. Indeed, I heard some were struggling with CS50 as their first course. 

I strongly recommend MITx6.00.1x. In the worst case (that it's too easy and you already know everything in it) you will lose just one week.

At the end of the day nobody can tell you what to do, it's your choice. You can even jump straight into an Advanced Application if you want! (Although you probably should not.) Start a course, see how things are going, check out the entire syllabus of the course, look at all the topics, and make your own decisions. Don't take my word for things either!

If you feel like your question is answered, please close the issue, thank you! Drop by [Gitter](https://gitter.im/open-source-society/computer-science), we can talk there directly!

**Comment by cauequeiroz** [comment]
@spamegg1 , wow, thank you so much for this answer!

I gave a careful look into MITx6.00.1x content (syllabus, problem sets and quizzes) and it seems more like "computer science stuff", different from Python for Everyone (which seems just a commercial-udemy-language/framework-course).

So I've decided to take this class before jump to core cs courses! I'll skip CS Tools because I already use these tools on a daily basis at my job.

Thanks again for the effort put in your answer, your opinion helped me a lot :)
